### PR TITLE
ST: Add tests for operator metrics

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

### Type of change

- Enhancement / new feature

### Description

This PR adds tests to cover metrics exposed by CO and UO.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci test_only profile=regression testcase=MetricsST,PrometheusST

**Comment by Frawless** [comment]
@strimzi-ci test_only profile=regression testcase=MetricsST,PrometheusST

**Comment by Frawless** [comment]
@strimzi-ci test_only profile=regression testcase=MetricsST,PrometheusST

**Comment by Frawless** [comment]
@strimzi-ci test_only profile=regression testcase=MetricsST,PrometheusST

**Comment by Frawless** [comment]
@strimzi-ci run test test_only profile=regression testcase=MetricsST,PrometheusST

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only profile=regression testcase=MetricsST,PrometheusST

File: systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java [hunk 1]
```diff
@@ -179,6 +187,77 @@ void testKafkaExporterDifferentSetting() throws InterruptedException, ExecutionE
...
+        assertCoMetricNotNull("strimzi_reconciliations_total", "KafkaUser");
+
+        Pattern connectResponse = Pattern.compile("strimzi_resources\\{kind=\"KafkaUser\",} ([\\d.][^\\n]+)");
+        ArrayList<Double> values = MetricsUtils.collectSpecificMetric(connectResponse, userOperatorMetricsData);
+        assertThat(values.stream().mapToDouble(i -> i).sum(), is((double) 2));
+    }
+
+    @Test
+    void testMirrorMaker2Metrics() {
+        Pattern connectResponse = Pattern.compile("kafka_connect_connect_worker_metrics_connector_count ([\\d.][^\\n]+)");
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java:252]: Pay attention at #2800 , it's going to change `kafka_connect_connect_worker_metrics_connector_count` to `kafka_connect_worker_connector_count`

File: systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java [hunk 2]
```diff
@@ -179,6 +187,77 @@ void testKafkaExporterDifferentSetting() throws InterruptedException, ExecutionE
...
+        ArrayList<Double> values = MetricsUtils.collectSpecificMetric(connectResponse, kafkaMirrorMaker2MetricsData);
+        assertThat(values.stream().mapToDouble(i -> i).sum(), is((double) 3));
+
+        connectResponse = Pattern.compile("kafka_connect_connect_worker_metrics_task_count ([\\d.][^\\n]+)");
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsST.java:256]: Ditto as above, it's going to be replaced by `kafka_connect_worker_task_count`

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only profile=regression testcase=PrometheusST

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only profile=prometheus testcase=PrometheusST

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only profile=metrics

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only profile=metrics

**Pull request merged**